core/modules/installer/installer.browser.pages.inc

```diff
@@ -0,0 +1,998 @@
+  $_SESSION['installer_browser_listed_projects'] = $results['projects'];
+
+  $list = array();
+  foreach ($results['projects'] as $project) {
+    $list[] = $project;
+  }
+
+  // Add the pager.
+  $total = $results['total'];
+  $items_per_page = 20;
+  $page = pager_default_initialize($total, $items_per_page);
+  $offset = $items_per_page * $page;
+  $start = ($total) ? $offset + 1 : 0;
+  $finish = $offset + $items_per_page;
+  if ($finish > $total) {
+    $finish = $total;
+  }
+
+  $sort_options = installer_browser_get_sort_options();
+  $current_order = isset($_SESSION['installer_browser_order_filter_' . $type]) ? $_SESSION['installer_browser_order_filter_' . $type] : 'score';
+  $current_sort = isset($_SESSION['installer_browser_sort_filter_' . $type]) ? $_SESSION['installer_browser_sort_filter_' . $type] : 'desc';;
+
+  $build = array();
+  $build['content'] = array(
+    'installer_browser_header' => array(
+      '#markup' => t('Showing @start to @finish of @total.', array(
+        '@start' => $start, '@finish' => $finish, '@total' => $total)),
+      '#weight' => 0,
+    ),
+    'installer_browser_sort_header' => array(
+      '#type' => 'item',
+      '#weight' => 2,
+      '#markup' => installer_browser_get_sort_widget($sort_options, $current_order, $current_sort),
+    ),
+    'installer_browser_list' => array(
+      '#markup' => theme('installer_browser_list', array('projects_list' => $list, 'type' => $type)),
+      '#weight' => 3,
+    ),
+    'pager' => array(
+      '#theme' => 'pager',
+      '#weight' => 99,
+    ),
+  );
+
+  return $build;
+}
+
+/**
+ * Page to display individual projects in a modal.
+ *
+ * @param $project
+ *   The project name.
+ *
+ * @return
+ *   Render array of a themed project.
+ */
+function installer_browser_project_display($project) {
+  $projects = $_SESSION['installer_browser_listed_projects'];
+  $form['project_dialog'] = array(
+    '#markup' => theme('installer_browser_project_dialog', array('project' => $projects[$project])),
+  );
+  return $form;
+}
+
+/**
+ * Page to install projects.
+ *
+ * @param $op
+ *    Operation to preform.
+ */
+function installer_browser_installation_page($op) {
+  backdrop_add_css(backdrop_get_path('module', 'installer') . '/css/installer.css', array('preprocess' => FALSE));
+
+  switch ($op) {
+    case 'select_versions':
+      backdrop_set_title(t("Select versions"));
+      $content = installer_browser_installation_select_versions_page();
+      break;
+    case 'install_dependencies':
+      backdrop_set_title(t("Install Dependencies"));
+      $content = installer_browser_installation_install_dependencies_page();
+      break;
+    case 'theme':
+      backdrop_set_title(t("Set themes"));
+      $content = installer_browser_installation_enable_page('theme');
+      break;
+    case 'enable':
+      backdrop_set_title(t("Enable modules"));
+      $content = installer_browser_installation_enable_page();
+      break;
+  }
```

Comment:
There's no validation on $op here, so I would add a `default:` case and return `MENU_NOT_FOUND`.